fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CTEContext.java

```diff
@@ -0,0 +1,168 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+package com.starrocks.sql.optimizer;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.common.FeConstants;
+import com.starrocks.sql.optimizer.base.ColumnRefSet;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class CTEContext {
+    // All CTE produce
+    private Map<String, OptExpression> produces;
```

Comment:
Why CTE id use string, not int?

Reply:
Use int is OK, there use string just for simplicity, can be modify it